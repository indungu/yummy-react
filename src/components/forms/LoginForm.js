import React from 'react';

const LoginForm = (props) => {
  return (
    <div>
      <form className="auth-form" onSubmit={props.onSubmit}>
        <div className="form-group">
          <label className="control-label">Email</label>
          <input
            type="email"
            name="email"
            className="form-control"
            onChange={props.onChange}
            autoComplete="current_email"
            placeholder="john.doe@user.com"
          />
        </div>
        <div className="form-group">
          <label className="control-label">Password</label>
          <input
            type="password"
            name="password"
            className="form-control"
            onChange={props.onChange}
            autoComplete="current_password"
            placeholder="Password"
          />
        </div>
        <div className="form-group">
          <button className="btn btn-primary btn-lg">
            Login
          </button>
        </div>
      </form>
    </div>
  );
};

export default LoginForm;
