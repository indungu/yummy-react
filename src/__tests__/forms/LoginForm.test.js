import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import LoginForm from '../../components/forms/LoginForm';

describe('Login', () => {
  const wrapper = shallow(<LoginForm />);
  it('Renders properly without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
