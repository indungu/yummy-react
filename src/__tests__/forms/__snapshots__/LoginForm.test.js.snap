// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Renders properly without crashing 1`] = `
<div>
  <form
    className="auth-form needs-validation"
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
      >
        Email
      </label>
      <input
        autoComplete="current_email"
        className="form-control"
        name="email"
        placeholder="john.doe@user.com"
        required={true}
        type="email"
      />
      <div
        className="valid-feedback"
      >
        Ok
      </div>
      <div
        className="invalid-feedback"
      >
        Please enter a valid email address.
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
      >
        Password
      </label>
      <input
        autoComplete="current_password"
        className="form-control"
        name="password"
        placeholder="Password"
        required={true}
        type="password"
      />
      <div
        className="valid-feedback"
      >
        Ok
      </div>
      <div
        className="invalid-feedback"
      >
        Please enter a password.
      </div>
    </div>
    <div
      className="form-group"
    >
      <button
        className="btn btn-primary btn-lg"
      >
        Login
      </button>
    </div>
  </form>
</div>
`;
